pontoon/base/models.py

```diff
@@ -2214,24 +2217,41 @@ def for_project_locale(
 
         # Filter by search parameters
         if search:
-            # https://docs.djangoproject.com/en/dev/topics/db/queries/#spanning-multi-valued-relationships
-            search_query = (search, locale.db_collation)
-            translation_matches = (
-                entities.filter(
-                    translation__string__icontains_collate=search_query,
-                    translation__locale=locale,
-                )
-                .values_list('id', flat=True)
-            )
-            entity_matches = (
-                entities.filter(
-                    Q(string__icontains=search) |
-                    Q(string_plural__icontains=search) |
-                    Q(comment__icontains=search) |
-                    Q(key__icontains=search)
-                )
-                .values_list('id', flat=True)
-            )
+            # Split search string on spaces except if between non-escaped quotes.
+            search_list = [
+                x.strip('"').strip("'").replace(UNUSABLE_SEARCH_CHAR, '"')
+                for x in re.findall(
+                    '([^\"]\\S*|\".+?\")\\s*',
+                    search.replace('\\"', UNUSABLE_SEARCH_CHAR))]
+
+            # Search for `"` and `'` when entered as search terms
+            if search == '"' and not search_list:
+                search_list = ['"']
+
+            if search == "'" and not search_list:
+                search_list = ["'"]
+
+            search_query_list = [(s, locale.db_collation) for s in search_list]
+
+            translation_filters = (
+                Q(translation__string__icontains_collate=search_query) &
```

Comment:
i generally try to avoid code style nits that are pep8 friendly, however...

according to pep8 you *should* put operators at end of line - but i really think this is wrong (a lot!)

my pref is to have the operators at beginning - like (using next lines as example) ...

```

            entity_filters = (
+                Q(string__icontains=search)
+                | Q(string_plural__icontains=search)
+                | Q(comment__icontains=search)
+                | Q(key__icontains=search)
+                for search in search_list)
+
````